### PR TITLE
fixing normalization to be region specific

### PR DESCRIPTION
**Pull request by mariajmolina**
normalization by region required some reorganization of the custom pytorch dataset -- note that random regions has been deprecated as a result of change

### CODE REVIEW

File: s2sml/torch_s2s_dataset.py [hunk 1]
```diff
@@ -277,13 +275,18 @@ def zscore_values(self, mnv, stdv):
...
-
-            self.std_val = xr.open_mfdataset(
-                self.list_of_era5, concat_dim='sample', combine='nested')[var].std(
-                skipna=True).values
+            tmp = xr.open_mfdataset(self.list_of_era5, concat_dim='sample', combine='nested')[var]
+            tmp = self.box_cutter(tmp)
+            
+            if not self.norm_pixel:
+                
+                self.mean_val = tmp.mean(skipna=True).values
```
Comment (katiedagon) [inline s2sml/torch_s2s_dataset.py]: Is this where we should do a weighted mean? See [this page](https://docs.xarray.dev/en/stable/examples/area_weighted_temperature.html) for a nice example using `xarray.DataArray.weighted`

File: s2sml/torch_s2s_dataset.py [hunk 2]
```diff
@@ -277,13 +275,18 @@ def zscore_values(self, mnv, stdv):
...
+                self.std_val = tmp.std(skipna=True).values
```
Comment (katiedagon) [inline s2sml/torch_s2s_dataset.py:288]: Question about how the standard deviation is calculated - is it over space and time?

**Review by kjmayer — APPROVED** [review]
Looks good to me!

File: s2sml/torch_s2s_dataset.py (hunk 2 above)
Comment (kjmayer) [reply s2sml/torch_s2s_dataset.py:288]: This is space and time, yeah.

File: s2sml/torch_s2s_dataset.py (hunk 1 above)
Comment (kjmayer) [reply s2sml/torch_s2s_dataset.py]: Oh right! I forgot about the weighting!

File: s2sml/torch_s2s_dataset.py (hunk 2 above)
Comment (mariajmolina) [reply s2sml/torch_s2s_dataset.py:288]: yup! space and time!

File: s2sml/torch_s2s_dataset.py (hunk 1 above)
Comment (mariajmolina) [reply s2sml/torch_s2s_dataset.py]: good point! we should add this -- so far, this should not be an issue if using small region, but should be changed to be weighted in case we want to train with larger region

File: s2sml/torch_s2s_dataset.py (hunk 2 above)
Comment (katiedagon) [reply s2sml/torch_s2s_dataset.py:288]: Thanks - so no weighting needed for the standard deviation calculation right? (I'm guessing no, but also trying to wrap my head around this for another project...)

Comment (mariajmolina) [reply s2sml/torch_s2s_dataset.py:288]: (note, that it is only across space and time for the time range selected, to avoid training data leakage)

Comment (mariajmolina) [reply s2sml/torch_s2s_dataset.py:288]: hmmm... I think we need to do weighting for std as well if we do it for mean

Comment (kjmayer) [reply s2sml/torch_s2s_dataset.py:288]: I was going to say the same as @mariajmolina, but I'm googling to double check.

File: s2sml/torch_s2s_dataset.py [hunk 3]
```diff
@@ -280,7 +280,11 @@ def zscore_values(self, mnv, stdv):
             
             if not self.norm_pixel:
                 
-                self.mean_val = tmp.mean(skipna=True).values
+                weights = np.cos(np.deg2rad(tmp.y))
```
Comment (mariajmolina) [inline s2sml/torch_s2s_dataset.py:283]: okay weighting added @katiedagon

**Review by katiedagon — APPROVED** [review]
